src/components/MovieTable/index.tsx

```diff
@@ -82,27 +90,29 @@ function MovieTable() {
   }, [curSortType, curTab]);
 
   const [showModal, setShowModal] = useState<boolean>(false);
+  const [showDeleteModal, setShowDeleteModal] = useState<boolean>(false);
   const [showPopup, setShowPopup] = useState<boolean>(false);
+  const onClickDelete = () => {
+    dispatch(deleteMovie(currentMovie.id));
+    setShowDeleteModal(false);
+    setShowPopup(true);
+  };
   const onClickMenuItem = (value: string, movie: Movie) => {
-    setCurrentMovie(movie);
+    dispatch(moviesAction.setCurrentMovie(movie));
     if (value === menuItem.EDIT) {
       setShowModal(true);
```

Comment:
I suggest you can put all the mode, modalVisible flag logic in Redux, then you just need one Movie Modal component in App level and been controlled by Redux state.